File: db4on/in/sharp/core/com/db4o/config/TClass.cs
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

using System;
using j4o.lang;
using com.db4o;

namespace com.db4o.config
{
	/// <exclude />
	public class TClass : ObjectConstructor
	{
		static readonly Class _stringClass = Class.GetClassForType(typeof(String));

		public void OnActivate(ObjectContainer objectContainer, object obj, object members)
		{
		}

		public Object OnInstantiate(ObjectContainer objectContainer, object obj)
		{
			try
			{
				return Class.ForName((String)obj);
			}
			catch (Exception exception)
			{
				return null;
			}
		}

		public Object OnStore(ObjectContainer objectContainer, object obj)
		{
			return ((Class)obj).GetName();
		}

		public Class StoredClass()
		{
			return _stringClass;
		}
	}
}

File: db4oj/core/src/com/db4o/config/TClass.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.config;

import com.db4o.ObjectContainer;

public class TClass implements ObjectConstructor
{
	public Object onStore(ObjectContainer con, Object object){
		return ((Class)object).getName();
	}
	
	public void onActivate(ObjectContainer con, Object object, Object members){
		// do nothing
	}

	public Object onInstantiate(ObjectContainer con, Object storedObject){
		try{
			// return Db4o.classForName((String)storedObject);
		    return Class.forName((String)storedObject);
		}catch(Exception e){}
		return null;
	}

	public Class storedClass(){
		return String.class;
	}
}
